ptgnn/neuralmodels/gnn/graphneuralnetwork.py

```diff
@@ -257,25 +323,39 @@ def __iterate_edge_types(
                 yield np.zeros(0, dtype=np.int32), np.zeros(0, dtype=np.int32)
 
     def tensorize(
-        self, datapoint: GraphData[TNodeData]
-    ) -> Optional[TensorizedGraphData[TTensorizedNodeData]]:
+        self, datapoint: GraphData[TNodeData, TEdgeData]
+    ) -> Optional[TensorizedGraphData[TTensorizedNodeData, TTensorizedEdgeData]]:
+        if len(datapoint.node_information) > self.max_nodes_per_graph:
+            self.LOGGER.warning("Dropping graph with %s nodes." % len(datapoint.node_information))
+            return None
+
+        if self.__edge_embedding_model is None:
+            tensorized_edge_features = None
+        else:
+            tensorized_edge_features = []
+            for edge_type in self.__edge_idx_to_type:
+                edge_features = datapoint.edge_features.get(edge_type)
+                if edge_features is None:
+                    tensorized_edge_features.append([])
```

Comment:
I'm a little bit confused by this one - the list is never consumed, right? Would it be better to use `None` here to make clear that this data is missing? (`[]` for "no data" can have weird effects when it's accessed using `for` or `zip` when it silently truncates instead of failing)

Reply:
Thanks, this was a bit confusing indeed. 

I've renamed `edge_features` to `edge_feature_for_edge_type`. `edge_features` is `None` when there are no edges of the given type in the datapoint, not when there are no edge features.

`tensorized_edge_features` should be None when features aren't used (= no `edge_embedding_model`)

Reply:
Ah! I indeed didn't get this, and the comment now helps as well.